codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -617,7 +617,7 @@ public ChannelFuture resetStream(final ChannelHandlerContext ctx, int streamId,
         }
 
         final ChannelFuture future;
-        if (stream.state() == IDLE || connection().local().created(stream)) {
```

Comment:
#4502 was necessary because we used to queue the initial headers frames and it was possible for the user to do a `writeHeaders()` (headers would be queued) and then `writeReset()` (reset would be sent, but remote doesn't know about the stream and responds with a protocol error and the connection is closed). Now that we don't queue headers this scenario should not be possible (stream is in idle and the remote peer doesn't know it) and we are now just trying to enforce the following:

https://tools.ietf.org/html/rfc7540#section-6.4
> RST_STREAM frames MUST NOT be sent for a stream in the "idle" state.
   If a RST_STREAM frame identifying an idle stream is received, the
   recipient MUST treat this as a connection error (Section 5.4.1) of
   type PROTOCOL_ERROR.

However this seems strange if the peer has sent a PRIORITY frame which creates a stream in the idle state ... we are not allowed to reset it ... and the peer may think it can later send frames on this stream ... however I think we now send a stream error in this case in [DefaultHttp2ConnectionDecoder](https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java#L562).